charts/argo-controller/templates/service.yaml

```diff
@@ -0,0 +1,24 @@
+{{- $metricsPort := .Values.prometheusMetrics.port }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Release.Name }}-metrics-service
+spec:
+  selector:
+    app.kubernetes.io/name: {{ .Release.Name }}
+  ports:
+    - port: {{ $metricsPort }}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ .Release.Name }}-metrics-service-monitor
+  labels:
+    label: {{ .Values.prometheusMetrics.label }}
```

Comment:
My creativity was lacking in naming this label. Let me know if anyone has thoughts on what it should be called. 

Also, `.Values.prometheusMetrics.label` is not defined in `values.yaml` because it sounds like it should be defined in monster-deploy.

Reply:
Maybe `monitorLabel` to make it clear that it's aimed at the service monitor?